### PR TITLE
identify, implement and test missing functions in MLX backend

### PR DESCRIPTION
**Pull request by Faisal-Alsrheed**
identify, implement, and test missing functions in the MLX backend.
at `keras/backend/mlx/math.py` and `keras/backend/mlx/numpy.py`
 
 

 I have added [mlx.core.meshgrid]( https://ml-explore.github.io/mlx/build/html/python/_autosummary/mlx.core.meshgrid.html#mlx.core.meshgrid
)
 
I have added [mlx.core.tensordot]( https://ml-explore.github.io/mlx/build/html/python/_autosummary/mlx.core.tensordot.html#mlx-core-tensordot) 
 
 
 I have added [mlx.core.round](https://ml-explore.github.io/mlx/build/html/python/_autosummary/mlx.core.round.html#mlx.core.round)
 
 
I have added some as TODOs.  I will be happy to address them later.

I have implemented and tested median
https://colab.research.google.com/drive/1QAnbnRQB-njkbMm8s852Go6LcOQH3ihN?usp=sharing

WIP almost done..

keras/backend/mlx/nn.py:conv
keras/backend/mlx/nn.py:depthwise_conv
keras/backend/mlx/nn.py:separable_conv
keras/backend/mlx/nn.py:conv_transpose

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
Thanks for the contribution!

**Comment by Faisal-Alsrheed** [comment]
Hi @fchollet,
Since there are conflicts, and to make it easier for you, I'm thinking of deleting this PR and creating a new updated one. 
Sound good?

File: keras/backend/mlx/numpy.py [hunk 1]
```diff
@@ -580,13 +580,33 @@ def maximum(x1, x2):
...
+        median_shape.insert(axis, 1)
+        median = median.reshape(median_shape)
+    return median
 
 
 def meshgrid(*x, indexing="xy"):
-    # TODO: Implement inline like linspace
-    raise NotImplementedError("The MLX backend doesn't support meshgrid yet")
+    # TODO: test with and without
+    # x = [convert_to_tensor(xi) for xi in x]
```
Comment (fchollet) [inline keras/backend/mlx/numpy.py:608]: Shouldn't we keep this line?

**Comment by fchollet** [comment]
> Since there are conflicts, and to make it easier for you, I'm thinking of deleting this PR and creating a new updated one.
Sound good?

Sure, sounds good!

**Pull request closed without merging**